plugins/woocommerce/includes/admin/class-wc-admin-meta-boxes.php

```diff
@@ -226,24 +226,31 @@ public function save_meta_boxes( $post_id, $post ) {
 	}
 
 	/**
-	 * Remove block-based templates from the list of available templates for products.
+	 * Remove irrelevant block templates from the list of available templates for products.
+	 * This will also remove custom created templates.
 	 *
 	 * @param string[] $templates Array of template header names keyed by the template file name.
 	 *
 	 * @return string[] Templates array excluding block-based templates.
 	 */
-	public function remove_block_templates( $templates ) {
-		if ( count( $templates ) === 0 || ! function_exists( 'gutenberg_get_block_template' ) ) {
+	public function filter_product_block_templates( $templates ) {
+		if ( count( $templates ) === 0 || ! wc_current_theme_is_fse_theme() || ! function_exists( 'gutenberg_get_block_template' ) ) {
```

Comment:
Not introduced by this PR, but `gutenberg_get_block_template` was [renamed](https://github.com/WordPress/wordpress-develop/blob/d085349cf5760294d1bdbc3f01e30a483d921f5d/src/wp-includes/block-template-utils.php#L750) to `get_block_template` in WP 5.9, so we might need to account for that name if Gutenberg is disabled. (I think it's ok doing that in a follow-up PR if you prefer)